### PR TITLE
BUG: UInt64Index.where with int64 value

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry

### CODE REVIEW

**Comment by simonjayhawkins** [comment]
bug? needs a release note. I would put in 1.3 for now. (it can be moved if needed after @jreback reviews)

using semver we can make backwards compatible bug fixes in patch releases but we tend to only fix regressions.

from https://semver.org/

Given a version number MAJOR.MINOR.PATCH, increment the:

MAJOR version when you make incompatible API changes,
MINOR version when you add functionality in a backwards compatible manner, and
PATCH version when you make backwards compatible bug fixes.

File: doc/source/whatsnew/v1.3.0.rst [hunk 1]
```diff
@@ -947,6 +947,8 @@ Numeric
 - Bug in :meth:`Series.count` would result in an ``int32`` result on 32-bit platforms when argument ``level=None`` (:issue:`40908`)
 - Bug in :class:`Series` and :class:`DataFrame` reductions with methods ``any`` and ``all`` not returning Boolean results for object data (:issue:`12863`, :issue:`35450`, :issue:`27709`)
 - Bug in :meth:`Series.clip` would fail if the Series contains NA values and has nullable int or float as a data type (:issue:`40851`)
+- Bug in :meth:`UInt64Index.where` and :meth:`UInt64Index.putmask` with an ``np.int64`` dtype ``other`` incorrectly raising integer ``other`` incorrectly raising ``TypeError`` (:issue:`41974`)
```
Comment (simonjayhawkins) [inline doc/source/whatsnew/v1.3.0.rst:950]:
```suggestion
- Bug in :meth:`UInt64Index.where` and :meth:`UInt64Index.putmask` with an ``np.int64`` dtype ``other`` incorrectly raising ``TypeError`` (:issue:`41974`)
```

**Comment by jreback** [comment]
this is fine for 1.3

**Comment by jreback** [comment]
@meeseeksdev backport 1.3.x

**Pull request merged**